### PR TITLE
Put restore point content upon new file creation

### LINKED ISSUE #394: Upload fails for external storage restore point location
Nextcloud version 24.0.3
External Storage and Backup apps are enabled. External storage is configured to a Local mount and Backup has a restore point location. 
`occ backup:point:upload` fails every time at `ExternalFolderService::updateMetadataFile`
 
https://github.com/nextcloud/backup/blob/0b1105fd161cdc22bb5101eeb0917531c7a546ac/lib/Service/ExternalFolderService.php#L533 It seems  deserializeJSON throws a TypeError which is not caught and the restore point file remains empty. Any subsequent runs of `occ backup:point:upload` fails to read the restore point.
A simple fix would be to initialize the restore point when the file is created.

### PR DESCRIPTION
**Pull request by sibiantony**
fix #394

### CODE REVIEW

**Comment by SkaiiNyght** [comment]
Implemented the file change on my system, it has worked at least for me. My previous error was:
An unhandled exception has been thrown:
TypeError: Argument 1 passed to OCA\Backup\Service\ExternalFolderService::deserialize() must be of the type array, null given, called in ...apps/backup/lib/Tools/Traits/TDeserialize.php:67
#0 ...apps/backup/lib/Tools/Traits/TDeserialize.php(141): OCA\Backup\Service\ExternalFolderService->deserialize()
#1 ...apps/backup/lib/Service/ExternalFolderService.php(219): OCA\Backup\Service\ExternalFolderService->deserializeJson()
#2 ...nc.skaiinyght.com/apps/backup/lib/Service/ExternalFolderService.php(174): OCA\Backup\Service\ExternalFolderService->getRestoringPoint()
#3 ...nc.skaiinyght.com/apps/backup/lib/Service/PointService.php(888): OCA\Backup\Service\ExternalFolderService->getRestoringPoints()
#4 ...apps/backup/lib/Command/PointList.php(122): OCA\Backup\Service\PointService->getRPFromInstances()
#5 ...3rdparty/symfony/console/Command/Command.php(255): OCA\Backup\Command\PointList->execute()
#6 ...core/Command/Base.php(168): Symfony\Component\Console\Command\Command->run()
#7 ...3rdparty/symfony/console/Application.php(1009): OC\Core\Command\Base->run()
#8 ...3rdparty/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#9 ...3rdparty/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun()
#10 ...lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run()
#11 ...console.php(100): OC\Console\Application->run()
#12 ...occ(11): require_once('...')
#13 sudo -u www-data ./occ backup:point:upload 20220820003913-full-AfDLb5g79tWcnaq

It would fail prior to initiating the health check

File: lib/Service/ExternalFolderService.php [hunk 1]
```diff
@@ -525,7 +525,7 @@ public function updateMetadataFile(
 				throw new MetadataException('metadata file not found');
 			}
 
-			$metadataFile = $folder->newFile(MetadataService::METADATA_FILE, '');
+			$metadataFile = $folder->newFile(MetadataService::METADATA_FILE, json_encode($point, JSON_PRETTY_PRINT));
```
Comment (ArtificialOwl) [inline lib/Service/ExternalFolderService.php:528]: can you also returns `$metadataFile` just after its creation ?

Comment (sibiantony) [reply lib/Service/ExternalFolderService.php]: Done.

**Review by ArtificialOwl — APPROVED** [review]

**Pull request merged**